src/controllers/hid/hidcontroller.h

```diff
@@ -89,7 +89,10 @@ class HidController final : public Controller {
     hid_device* m_pHidDevice;
     HidControllerPreset m_preset;
 
-    unsigned char m_pPollData[255];
+    static constexpr int m_iNumBuffers = 2;
+    static constexpr int m_iBufferSize = 255;
+    unsigned char m_pPollData[m_iNumBuffers][m_iBufferSize];
```

Comment:
Does C++ initialize these buffers to 0? On first run, whille the comparison with the previous buffer compare with uninitialized memory? Not sure if this is an issue, I suspect memcmp will just return a nonzero value.

Also, do we need to reset this buffer when the controller is closed? Otherwise I think memcmp would return 0 after re-enabling the controller and the script won't receive input until you press a button or move a fader on the controller. I suspect that this might lead to softtakeover issues, but I'm not sure.

Reply:
I made it zero the buffers in HidController::open a2e3267

Reply:
> Does C++ initialize these buffers to 0?

Only global memory is initialized with 0. In the context of a member variable you cannot be sure, because you don't know in which memory the object is instantiated.